File: scripts/_archive/clean_imports.py
import os

CONTENT_DIR = "src/content/projects"
IMPORTS_TO_REMOVE = [
    "import { YouTube } from '@astro-community/astro-embed-youtube';",
    "import ModelViewer from '@components/mdx/ModelViewer.astro';"
]

def clean_imports():
    count = 0
    for filename in os.listdir(CONTENT_DIR):
        if not filename.endswith(".mdx"): continue
        
        path = os.path.join(CONTENT_DIR, filename)
        with open(path, "r", encoding="utf-8") as f:
            lines = f.readlines()
            
        new_lines = []
        changed = False
        for line in lines:
            stripped = line.strip()
            if stripped in IMPORTS_TO_REMOVE:
                changed = True
                continue # Skip this line
            new_lines.append(line)
            
        if changed:
            with open(path, "w", encoding="utf-8") as f:
                f.writelines(new_lines)
            print(f"cleaned: {filename}")
            count += 1
            
    print(f"Total files cleaned: {count}")

if __name__ == "__main__":
    clean_imports()

File: scripts/_archive/clean_imports.js
import fs from "fs";
import path from "path";

const PROJECTS_DIR = path.join(process.cwd(), "src/content/projects");

function cleanProjectFiles() {
	if (!fs.existsSync(PROJECTS_DIR)) {
		console.error(`Directory not found: ${PROJECTS_DIR}`);
		return;
	}

	const projects = fs.readdirSync(PROJECTS_DIR);
	let modifiedCount = 0;

	projects.forEach((projectDir) => {
		const dirPath = path.join(PROJECTS_DIR, projectDir);
		if (!fs.statSync(dirPath).isDirectory()) return;

		const mdxPath = path.join(dirPath, "index.mdx");
		if (fs.existsSync(mdxPath)) {
			const content = fs.readFileSync(mdxPath, "utf8");
			const lines = content.split("\n");
			const newLines = [];
			let hasChanges = false;

			for (let i = 0; i < lines.length; i++) {
				const line = lines[i];
				const trimmed = line.trim();

				// Check for imports to remove
				if (trimmed.startsWith("import { YouTube }") || trimmed.startsWith("import ModelViewer")) {
					hasChanges = true;
					continue; // Skip this line
				}

				newLines.push(line);
			}

			if (hasChanges) {
				const newContent = newLines.join("\n");
				fs.writeFileSync(mdxPath, newContent);
				console.log(`Cleaned imports from ${projectDir}`);
				modifiedCount++;
			}
		}
	});

	console.log(`\nCleanup complete. Modified ${modifiedCount} files.`);
}

cleanProjectFiles();
